src/valkey/valkey.zig

```diff
@@ -674,6 +679,36 @@ pub const ValkeyClient = struct {
             return;
         }
 
+        // Handle initial SELECT response
+        if (this.flags.is_selecting_db_internal) {
+            this.flags.is_selecting_db_internal = false;
+
+            switch (value.*) {
+                .Error => |err_str| {
+                    this.fail(err_str, protocol.RedisError.InvalidCommand);
+                    return;
+                },
+                .SimpleString => |ok_str| {
+                    if (std.mem.eql(u8, ok_str, "OK")) {
+                        // SELECT was successful.
+                        debug("SELECT {d} successful", .{this.database});
+                        // Connection is now fully ready on the specified database.
+                        // If any commands were queued while waiting for SELECT, try to send them.
+                        this.sendNextCommand();
+                    } else {
+                        // SELECT returned something other than "OK"
+                        this.fail("SELECT command failed with non-OK response", protocol.RedisError.InvalidResponse);
+                        return;
+                    }
+                },
+                else => { // Unexpected response type for SELECT
+                    this.fail("Received non-SELECT response while in the SELECT state.", protocol.RedisError.InvalidResponse);
+                    return;
+                },
+            }
+            return; // SELECT response handled, do not proceed to user command logic
```

Comment:
nit: this could be `return switch(...` or only one `return` after the switch. might make it a little more clear